eodatasets3/images.py

```diff
@@ -686,6 +683,64 @@ def create_thumbnail(
                                 index,
                             )
 
+    def create_thumbnail_singleband(
+        self,
+        in_file: Path,
+        out_file: Path,
+        bit: int = None,
+        lookup_table: Dict[int, Tuple[int, int, int]] = None,
+    ):
+        """
+        Write out a JPG thumbnail from a singleband image.
+        This takes in a path to a valid raster dataset and writes
+        out a file with only the values of the bit (integer) as white
+        """
+        assert not (
+            bit is not None and lookup_table is not None
+        ), "Please set either bit or lookup_table, and not both of them"
```

Comment:
Assertions are quietly disabled if the user runs with `python -O` or `PYTHONOPTIMIZE=true`  (I've seen some Linux distros do this by default). It's uglier, but explicit is safer:
```suggestion
        if bit is not None and lookup_table is not None:
            raise ValueError(
                "Please set either bit or lookup_table, and not both of them"
            )
```